piff/optical_model.py

```diff
@@ -175,52 +268,211 @@ def fit(self, star):
+
+        :returns: a galsim.GSObject instance
+        """
+        if self.atmo_type == 'VonKarman':
+            if L0 is None or L0==0.:
+                raise ValueError("No value specified for VonKarman L0 ")
+            else:
+                atm = galsim.VonKarman(lam=self.lam, r0=r0, L0=L0, flux=1.0, gsparams=self.gsparams)
+        elif self.atmo_type == 'Kolmogorov':
+            atm = galsim.Kolmogorov(lam=self.lam, r0=r0, flux=1.0, gsparams=self.gsparams)
+        elif self.atmo_type == 'None' or self.atmo_type == None:
+            atm = galsim.DeltaFunction()
         else:
-            prof = galsim.Convolve(prof)
+            raise ValueError("Invalid atmo_type ",self.atmo_type)
+
+        # shear
+        if g1!=0.0 or g2!=0.0:
+            atm = atm.shear(g1=g1, g2=g2)
+        return atm
+
+    def unpack_params(self,params):
+        """Unpack parameter values
+
+        :param params               An ndarray or list with parameters in order
+
+        :returns: zernike_coeff,r0,L0,g1,g2
+        """
+        kwargs = self.params_tokwargs(params)
+        return kwargs['zernike_coeff'],kwargs['r0'],kwargs['L0'],kwargs['g1'],kwargs['g2']
+
+    def params_tokwargs(self,params):
+        """Fill kwarg from params ndarray or list
+
+        :param params               An ndarray or list with parameters in order
+
+        :returns: a dictionary of named parameters
+        """
+        kwargs = {}
+        kwargs['zernike_coeff'] = params[self.idx_z0:self.idx_z0+self.nZ+1]
+        kwargs['r0'] = params[self.idx_r0]
+        kwargs['L0'] = params[self.idx_L0]
+        kwargs['g1'] = params[self.idx_g1]
+        kwargs['g2'] = params[self.idx_g2]
+        return kwargs
+
+    def kwargs_toparams(self,zernike_coeff=[0,0,0,0,0],r0=0.15,L0=0.,g1=0.,g2=0.):
+        """Fill params ndarray from kwargs
+
+        :param zernike_coeff:      A ndarray or list with [0, 0, z2, z3, z4, z5, z6...z11..z37] [default=[0,0,0,0,0]]
+        :param r0:                 Value of r0 for Atmospheric Kernel. [default=0.15]
+        :param L0:                 Value of L0 for Atmospheric Kernel. [default=0.0]
+        :param g1:                 Value of g1 to Shear PSF [default=0.]
+        :param g2:                 Value of g2 to Shear PSF [default=0.]
+
+        :returns: an ndarray of parameters
+        """
+        params = np.zeros(self.param_len)
+
+        # adapt to a zernike_coeff array shorter or longer than 0:37+1
+        zlen = len(zernike_coeff)
+        if zlen<=self.nZ+1:
+            params[self.idx_z0:self.idx_z0+zlen] = zernike_coeff
+        elif zlen>self.nZ+1:
+            params[self.idx_z0:self.idx_z0+self.nZ+1] = zernike_coeff[0:self.nZ+1]
+
+        params[self.idx_r0] = r0
+        params[self.idx_L0] = L0
+        params[self.idx_g1] = g1
+        params[self.idx_g2] = g2
+        return params
+
+
+    def getProfile(self,*params,zernike_coeff=None,r0=None,L0=None,g1=None,g2=None):
```

Comment:
This is the wrong signature for this function.  This is called by the base class, so it should always just be `getProfile(params)`.